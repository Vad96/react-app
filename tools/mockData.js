const courses = [
  {
    id: 1,
    title: "Securing React Apps with Auth0",
    slug: "react-auth0-authentication-security",
    authorId: 1,
    category: "JavaScript"
  },
  {
    id: 2,
    title: "React: The Big Picture",
    slug: "react-big-picture",
    authorId: 1,
    category: "JavaScript"
  },
  {
    id: 3,
    title: "Creating Reusable React Components",
    slug: "react-creating-reusable-components",
    authorId: 1,
    category: "JavaScript"
  },
  {
    id: 4,
    title: "Building a JavaScript Development Environment",
    slug: "javascript-development-environment",
    authorId: 1,
    category: "JavaScript"
  },
  {
    id: 5,
    title: "Building Applications with React and Redux",
    slug: "react-redux-react-router-es6",
    authorId: 1,
    category: "JavaScript"
  }
];

const authors = [
  { id: 1, name: "James Hetfield" },
  { id: 2, name: "Tom Araya" },
  { id: 3, name: "Dave Mustane" }
];

const newCourse = {
  id: null,
  title: "",
  authorId: null,
  category: ""
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newCourse,
  courses,
  authors
};
