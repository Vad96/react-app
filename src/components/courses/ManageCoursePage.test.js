import React from "react";
import renderer from "react-test-renderer";
import { authors, courses } from "../../../tools/mockData";
import { ManageCoursePage } from "./ManageCoursePage";

const props = {
authors,
courses,
history,
saveCourse: jest.fn(),
loadAuthors: jest.fn(),
loadCourses: jest.fn(),
};

it("should contain form with 3 input fields", () => {
  const tree = renderer.create(
      <ManageCoursePage {...props} />
    );
  
    expect(tree).toMatchSnapshot();
});
  