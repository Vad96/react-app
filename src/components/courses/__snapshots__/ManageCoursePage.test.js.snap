// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly after clicking submit button 1`] = `
<form>
  <h2>
    Add
     Course
  </h2>
  <div
    className="form-group has-error"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[Function]}
        type="text"
      />
      <div
        className="alert alert-danger"
      >
        Title is required.
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="authorId"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Select Author
        </option>
        <option
          value={1}
        >
          James Hetfield
        </option>
        <option
          value={2}
        >
          Tom Araya
        </option>
        <option
          value={3}
        >
          Dave Mustane
        </option>
      </select>
      <div
        className="alert alert-danger"
      >
        Author is required
      </div>
    </div>
  </div>
  <div
    className="form-group has-error"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="category"
        onChange={[Function]}
        type="text"
      />
      <div
        className="alert alert-danger"
      >
        Category is required
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onSubmit={[Function]}
    type="submit"
  >
    Save
  </button>
</form>
`;
